added fixes to allow services to start up with docker-compose up

Donovan-Medina opened the pull request:
According to the [docs](https://go.dev/doc/install), you must export the path for go in order for it to be seen.

Donovan-Medina commented:
From lines 13 to 20, I feel like these could be environment vars that would be exported before from a .env file. If the Config repo is something that could change constantly, or the org/user is going to change.

Donovan-Medina commented:
From lines 13 to 20, I feel like these could be environment vars that would be exported before from a .env file. If the Config repo is something that could change constantly, or the org/user is going to change.

uwardlaw commented on control-services/bin/setup.sh:
```diff
@@ -26,6 +26,7 @@ sudo apt-get upgrade -y
 sudo rm -rf /usr/local/go
 wget https://go.dev/dl/go1.17.7.linux-amd64.tar.gz -P /tmp/
 sudo tar -C /usr/local/ -xzf /tmp/go1.17.7.linux-amd64.tar.gz
+export PATH=$PATH:/usr/local/go/bin
```
Does this need to happen at line 29?

At the bottom:
`echo 'export PATH="$PATH:/usr/local/go/bin:/home/control/go/bin' | sudo tee -a /etc/profile`

Donovan-Medina replied:
I didn't catch that in the bottom, but yes it needs to happen before calling "go version". So perhaps just moving what's from the bottom and putting it on line 29. I can refractor that today.

Donovan-Medina replied:
There is another possibility of removing the "check go version" around line 30, or placing that after the path has been exported so it doesn't completely fail out of the setup.sh script.

Another clarifying question, why are we adding `home/control/go/bin` to `$PATH`? when the `go/bin` is in `/usr/local/`?

Donovan-Medina replied:
Added Comment of overall changes. Removed the line `go version`.

Donovan-Medina commented:
- Removed `go version` from vater/control-services/bin/setup.sh that caused an issue since the PATH is not exported until the bottom of the setup.sh file. There is no usage from the output of go version
- Added semaphore version to vater/control-services/images/semaphore/dockerfile. Until there is a ENV var we are wanting to add in the future. No other service is using a environment var to change the version type
- Added a .env.example so future devs can know what vars should be put inside a .env file
- Added a casc_configs directory in vater/control-services/images/jenkins/ that is passed as a volume for docker-compose.yml to use. This is needed as an ENV var for Jenkins plugin Configuration as Code to work.

marissaeinhorn commented on control-services/.env.example:
```diff
@@ -0,0 +1,9 @@
+
+gitea_db_user=gitea
+gitea_db_password=gitea
+semaphore_admin_password=cangetin
```
These credentials are held in `config.json`.  We don't need to put credentials in another place and cause confusion whenever we change passwords/jump networks.  I dont see how this env file will inform future devs unless your recommending a design change to the existing structure of storing/referencing service credentials?

Donovan-Medina replied:
Credentials, if stored in config.json or .env should not be committed to the repo.
Is there any documentation for using `config.json`? I've only ever seen [docker-compose .env](https://docs.docker.com/compose/environment-variables/#the-env-file)

marissaeinhorn replied:
Not the most up to date version but heres documentation https://github.com/dev4vater/vater/wiki/Configuration-File

marissaeinhorn commented on control-services/bin/setup.sh:
```diff
@@ -27,9 +27,6 @@ sudo rm -rf /usr/local/go
 sudo tar -C /usr/local/ -xzf /tmp/go1.17.7.linux-amd64.tar.gz
 
-# Check go version
-go version
```
Move this to the bottom after you modify the path.  Its a good debugging tool/status check to have in setup

Donovan-Medina replied:
Moved.

marissaeinhorn commented on control-services/images/semaphore/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM ansiblesemaphore/semaphore:dynamicVars
```
This will break our current semaphore build.  We have forked the repo and built a docker container with custom changes.  Within that build we reference a specific version of semaphore

Donovan-Medina replied:
Reverted. How is that custom image being bundled with VATER?
